cfme/tests/cloud/test_quota.py

```diff
@@ -298,3 +305,195 @@ def test_project_quota_enforce_via_lifecycle_cloud(
+    field, value = request.param
+    roottenant = appliance.collections.tenants.get_root_tenant()
+    roottenant.set_quota(**{'{}_cb'.format(field): True, field: value})
+    yield
+    roottenant.set_quota(**{'{}_cb'.format(field): False})
+
+
+@pytest.fixture(scope="module")
+def dialog(appliance, automate_flavour_method):
+    """This fixture is used to create dynamic service dialog"""
+    data = {
+        "buttons": "submit,cancel",
+        "label": "flavour_dialog_{}".format(fauxfactory.gen_alphanumeric()),
+        "dialog_tabs": [
+            {
+                "display": 'edit',
+                "label": "New Tab",
+                "position": 0,
+                "dialog_groups": [
+                    {
+                        "display": "edit",
+                        "label": "New section",
+                        "position": 0,
+                        "dialog_fields": [
+                            {
+                                "name": "option_0_instance_type",
+                                "description": "flavor_dialog",
+                                "data_type": "string",
+                                "display": "edit",
+                                'display_method_options': {},
+                                'required_method_options': {},
+                                'default_value': '',
+                                'values_method_options': {},
+                                "required": True,
+                                "label": "instance_type",
+                                "dynamic": True,
+                                'show_refresh_button': True,
+                                'load_values_on_init': True,
+                                'read_only': False,
+                                'auto_refresh': False,
+                                'visible': True,
+                                "type": "DialogFieldDropDownList",
+                                "resource_action": {
+                                    "resource_type": "DialogField",
+                                    "ae_namespace": automate_flavour_method.namespace.name,
+                                    "ae_class": automate_flavour_method.klass.name,
+                                    "ae_instance": automate_flavour_method.name,
+                                },
+                            },
+                        ],
+                    }
+                ],
+            }
+        ],
+    }
+    dialog_rest = appliance.rest_api.collections.service_dialogs.action.create(**data)[0]
+    yield appliance.collections.service_dialogs.instantiate(label=dialog_rest.label)
+    dialog_rest.action.delete()
+
+
+def get_quota_message(request, appliance, catalog, catalog_item_name, dialog_values=None):
+    """Returns the quota requested by particular type of flavor type"""
+    service_catalogs = ServiceCatalogs(appliance, catalog, catalog_item_name, dialog_values)
```

Comment:
This class _really_ needs to be converted to collection+entities, along with MyService.